### ISSUE TITLE
HTTP 400

### ISSUE THREAD

**Issue by grafpb** [OriginStampTimestamping/originstamp-client-python#1]
I used the example code from the documentation as shown below. This results in ApiExecption (400) error.

```python
api_instance = originstamp_client.TimestampApi()
authorization = API_KEY
notification = originstamp_client.Notification(currency=0, notification_type=0, target=EMAIL)
timestamp_request = originstamp_client.TimestampRequest(comment=COMMENT, hash=h, notifications=notification)  # TimestampRequest | DTO for the hash submission. Add all relevant information concerning your hash submission.
    try:
        # Submission
        api_response = api_instance.create_timestamp(authorization, timestamp_request)
    except ApiException as e:
        print("Exception when calling TimestampApi->create_timestamp: %s\n" % e)
```
results in

```
Exception when calling TimestampApi->create_timestamp: (400)
Reason: 
HTTP response headers: HTTPHeaderDict({'Expect-CT': 'max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"', 'Content-Length': '0', 'Server': 'cloudflare', 'Date': 'Mon, 06 May 2019 10:45:43 GMT', 'CF-RAY': '4d2a5dc3c98497e4-FRA', 'Connection': 'keep-alive', 'Set-Cookie': '__cfduid=dafd9f66a2e3bc643ac630b2887278f241557139543; expires=Tue, 05-May-20 10:45:43 GMT; path=/; domain=.originstamp.com; HttpOnly'})
```

i copied the data used inside the api to run a request with curl which gives the same error

```
curl -d "{'notifications': {'notification_type': 0, 'currency': 0, 'target': '***'}, 'comment': 'test', 'hash': '53a31bc8d49c23f54116ca2786f4f78bfa8e64b08a95a6e4e6981cec240053c1'}" -H "Authorization: ****" -H "Content-Type: application/json" -H"User-Agent: Swagger-Codegen/1.0.0/python" -H"Accept: application/json" -v  https://api.originstamp.com/v3/timestamp/create
```

```
*   Trying 104.18.38.26...
* Connected to api.originstamp.com (104.18.38.26) port 443 (#0)
* found 148 certificates in /etc/ssl/certs/ca-certificates.crt
* found 597 certificates in /etc/ssl/certs
* ALPN, offering http/1.1
* SSL connection using TLS1.2 / ECDHE_ECDSA_AES_128_GCM_SHA256
* 	 server certificate verification OK
* 	 server certificate status verification SKIPPED
* 	 common name: sni190452.cloudflaressl.com (matched)
* 	 server certificate expiration date OK
* 	 server certificate activation date OK
* 	 certificate public key: EC
* 	 certificate version: #3
* 	 subject: OU=Domain Control Validated,OU=PositiveSSL Multi-Domain,CN=sni190452.cloudflaressl.com
* 	 start date: Fri, 03 May 2019 00:00:00 GMT
* 	 expire date: Sat, 09 Nov 2019 23:59:59 GMT
* 	 issuer: C=GB,ST=Greater Manchester,L=Salford,O=COMODO CA Limited,CN=COMODO ECC Domain Validation Secure Server CA 2
* 	 compression: NULL
* ALPN, server accepted to use http/1.1
> POST /v3/timestamp/create HTTP/1.1
> Host: api.originstamp.com
> Authorization: ****
> Content-Type: application/json
> User-Agent: Swagger-Codegen/1.0.0/python
> Accept: application/json
> Content-Length: 187
> 
* upload completely sent off: 187 out of 187 bytes
< HTTP/1.1 400 
< Date: Mon, 06 May 2019 10:39:36 GMT
< Content-Length: 0
< Connection: keep-alive
< Set-Cookie: __cfduid=d162c80df244f3b1002b0370aacecd90f1557139176; expires=Tue, 05-May-20 10:39:36 GMT; path=/; domain=.originstamp.com; HttpOnly
< Expect-CT: max-age=604800, report-uri="https://report-uri.cloudflare.com/cdn-cgi/beacon/expect-ct"
< Server: cloudflare
< CF-RAY: 4d2a54cceb13c2fe-FRA
< 
* Connection #0 to host api.originstamp.com left intact
```

**Comment by thhepp** [comment]
Thank you for feedback. We will have a look at it.

I am not able to execute your cURL, because of quotation marks + incorrect request body (notifications array):

`curl -X POST 'https://api.originstamp.com/v3/timestamp/create' -H  'accept: application/json' -H  'Authorization: ***' -H  'Content-Type: application/json' -d '{"hash": "53a31bc8d49c23f54116ca2786f4f78b9a8e64b08a95a6e4e6981cec240053c1", "notifications": [{"currency": 0, "notification_type": 0, "target": "***"}]}'
`

Nevertheless, you are facing problems with the python client. Please give us a few days for further investigations.

**Comment by danmuf** [comment]
Concerning your issue, I would like to refer to the docs for the TimestampRequest model ([link here](https://github.com/OriginStampTimestamping/originstamp-client-python/blob/master/docs/TimestampRequest.md)). There you can see that you have to pass a **list** of notifications to the model but you only added `notifications=notification`. So to fix this issue, just replace the notifications parameter with `notifications=[notification]` in the model constructor. 

Additionally, I've released version 1.0.2 which includes a fix for Python 3.7. 
If the above solution does not work for you, please let us know and we will have another look at it.

**Comment by grafpb** [comment]
Thanks. Using `notifications=[notification]` now works for me!
